share/spice/color_picker/color_picker.js

```diff
@@ -0,0 +1,767 @@
+                value = current_color.hsv.value;
+
+            hue = to_bounded_integer(hue, 0, 359);
+            
+            update_all_from_hsv(hue, saturation, value);
+        }
+
+        function saturation_change() {
+            var hue = current_color.hsv.hue,
+                saturation = local_dom.$saturation_input.val(),
+                value = current_color.hsv.value;
+
+            saturation = to_bounded_integer(saturation, 0, 100);
+
+            update_all_from_hsv(hue, saturation, value);
+        }
+
+        function value_change() {
+            var hue = current_color.hsv.hue,
+                saturation = current_color.hsv.saturation,
+                value = local_dom.$value_input.val();
+
+            value = to_bounded_integer(value, 0, 100);
+            
+            update_all_from_hsv(hue, saturation, value);
+        }
+
+        function cyan_change() {
+            var cyan = local_dom.$cyan_input.val(),
+                magenta = current_color.cmyk.magenta,
+                yellow = current_color.cmyk.yellow,
+                black = current_color.cmyk.black;
+
+            cyan = to_bounded_number(cyan, 0, 100);
+            
+            update_all_from_cmyk(cyan, magenta, yellow, black);
+        }
+
+        function magenta_change() {
+            var cyan = current_color.cmyk.cyan,
+                magenta = local_dom.$magenta_input.val(),
+                yellow = current_color.cmyk.yellow,
+                black = current_color.cmyk.black;
+
+            magenta = to_bounded_number(magenta, 0, 100);
+            
+            update_all_from_cmyk(cyan, magenta, yellow, black);
+        }
+
+        function yellow_change() {
+            var cyan = current_color.cmyk.cyan,
+                magenta = current_color.cmyk.magenta,
+                yellow = local_dom.$yellow_input.val(),
+                black = current_color.cmyk.black;
+
+            yellow = to_bounded_number(yellow, 0, 100);
+            
+            update_all_from_cmyk(cyan, magenta, yellow, black);
+        }
+
+        function black_change() {
+            var cyan = current_color.cmyk.cyan,
+                magenta = current_color.cmyk.magenta,
+                yellow = current_color.cmyk.yellow,
+                black = local_dom.$black_input.val();
+
+            black = to_bounded_number(black, 0, 100);
+            
+            update_all_from_cmyk(cyan, magenta, yellow, black);
+        }
+
+        function hex_change() {
+            var hex = local_dom.$hex_input.val();
+
+            //There are a few ways a new hex string could look and still be valid. It may more may
+            //  not start with a '#' character, and it may contain some combination of uppercase and
+            //  lowercase letters. It can also contain either three or six hex numerals. Any other
+            //  number and the string is either too long, or ambiguous (e.g. #abc -> #0a0b0c, but
```

Comment:
> e.g. #abc -> #0a0b0c

Yeah, as noted before, [this should be `#aabbcc`](http://www.websiteoptimization.com/speed/tweak/hex/).